src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/JwtAuthenticationFilter.java

```diff
@@ -43,11 +50,32 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
         filterChain.doFilter(request, response);
     }
 
+    private void addRefreshTokenToResponse(
+        HttpServletResponse response, String jwt, UsernamePasswordAuthenticationToken authentication
+    ) {
+        Date newDate = null;
+        try {
+            newDate = tokenProvider.getMaxExpiryDateFromJwt(jwt);
```

Comment:
so how many times is during one request this token parsed and encrypted... this is 3rd I think. This should be improved so the token and claims are only encrypted once and values reused rather than parsed each time a claim is needed.